packages/debugger/lib/controller/sagas/index.js

```diff
@@ -205,10 +206,15 @@ function* stepOver() {
     // either: function depth is greater than starting (ignore function calls)
     // or, if we're at the same depth, keep stepping until we're on a new
     // line (which may be in a new file)
+    // or if we're at the declaration of a function, keep stepping if the
+    // function depth is greater than or equal to starting, i.e. we're inside
+    // the function, which is already required above. Consequently, we're past
+    // the end of the function.
     (currentDepth > startingDepth ||
       (currentLocation.source.id === startingLocation.source.id &&
         currentLocation.sourceRange.lines.start.line ===
-          startingLocation.sourceRange.lines.start.line))
+          startingLocation.sourceRange.lines.start.line) ||
+      startingNode.nodeType === "FunctionDefinition")
```

Comment:
(To clarify, `startingNode` may be `null` because you're in unmapped code; or the debugger is lacking the source for your current position; or you're debugging Vyper rather than Solidity or Yul...)

Reply:
Actually, thought I just had: You might want to also account here for the case of `YulFunctionDefinition`?  That might be a little tricky though, Yul can be messy and its function definitions don't necessarily work as cleanly as Solidity's do.  I think it's doable, but it's *probably* fine to leave that out for now.... but if you do want to include it you should probably ask me for some help there, I can give you some pointers about handling those properly. :)